lib/internal/assert.js

```diff
@@ -49,22 +54,53 @@ function inspectValue(val) {
 }
 
 function createErrDiff(actual, expected, operator) {
-  var other = '';
-  var res = '';
-  var lastPos = 0;
-  var end = '';
-  var skipped = false;
+  let other = '';
+  let res = '';
+  let lastPos = 0;
+  let end = '';
+  let skipped = false;
   const actualLines = inspectValue(actual);
   const expectedLines = inspectValue(expected);
-  const msg = READABLE_OPERATOR[operator] +
-        `:\n${green}+ expected${white} ${red}- actual${white}`;
+  const msg = kReadableOperator[operator] +
+        `:\n${green}+ actual${white} ${red}- expected${white}`;
   const skippedMsg = ` ${blue}...${white} Lines skipped`;
 
+  let i = 0;
+  let indicator = '';
+
+  // If "actual" and "expected" fit on a single line and they are not strictly
+  // equal, check further special handling.
+  if (actualLines.length === 1 && expectedLines.length === 1 &&
+    actualLines[0] !== expectedLines[0]) {
+    const inputLength = actualLines[0].length + expectedLines[0].length;
+    // If the character length of "actual" and "expected" together is less than
+    // kMaxShortLength and if neither is an object and at least one of them is
+    // not `zero`, use the strict equal comparison to visualize the output.
+    if (inputLength <= kMaxShortLength) {
+      if ((typeof actual !== 'object' || actual === null) &&
+          (typeof expected !== 'object' || expected === null) &&
+          (actual !== 0 || expected !== 0)) { // -0 === +0
+        return `${kReadableOperator[operator]}:\n\n` +
+            `${actualLines[0]} !== ${expectedLines[0]}\n`;
+      }
+    // If the stdout is a tty and the input length is lower than the current
+    // columns per line, add a mismatch indicator below the output.
+    } else if (process.stdout.isTTY &&
+               inputLength < process.stdout.columns) {
```

Comment:
I’d prefer not to add a dependency on `process.stdout` here … there’s no guarantee that this is where the error output ends up, and it makes `assert` depend on something it shouldn’t depend on (or makes it more so, at least).

Can we pick a value, e.g. 80 or 100 or so?

Reply:
I am aware that there is no guarantee that this is where the error output ends up. It is also the reason why I only added the indicator if it is a tty (and yes, even then the error could end up somewhere else but that is unlikely).

If I understand you correct you would actually like to have the indicator all the time? In that case I could add a default of 80 in case it is not a tty and check stdout otherwise?

That all aside: I would actually like to add a proper diffing algorithm at some point with which it would be possible to use colors to highlight the differences instead. But that is a bit more work and I got quite some pushback when it comes to improvements to assert, so I am not really sure if that is something worth trying.